### PR TITLE
feat(js): shared library

### PR DESCRIPTION
**Pull request by blu3beri**
- Shared interface library between the NodeJS side and the React Native side. Some minor TODO's are still included and will be resolved before merging.
- Does not include actual functionality, those will come in future PRs.
- (this is not going to main, but `javascript-wrapper`.

Signed-off-by: Berend Sliedrecht <berend@animo.id>

### CODE REVIEW

File: wrappers/javascript/shared/src/types/IndyVdr.ts [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+  PoolSubmitRequestOptions,
+  RequestSetEndorserOptions,
+  RequestSetMultiSignatureOptions,
+  RequestSetTxnAuthorAgreementAcceptanceOptions,
+} from '../indyVdr'
+import type { PrepareTxnAuthorAgreementAcceptanceOptions, RequestSetSignatureOptions } from './builderTypes'
+import type { PoolStatus, SubmitAction, SubmitRequest, Transactions, Verifiers } from './types'
+
+// TODO: proper documentation
+export interface IndyVdr {
```
Comment (TimoGlastra) [inline wrappers/javascript/shared/src/types/IndyVdr.ts:39]: Maybe we can group the methods in this interface to make them a bit more readbable? so all `buildXXX` below each other without newline, etc...

File: wrappers/javascript/shared/src/utils/checkTypes.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+type ShapeOf<T> = {
+  [Property in keyof T]: T[Property]
+}
+
+export type GenericMethods<Base extends { [method: string]: any }> = {
```
Comment (TimoGlastra) [inline wrappers/javascript/shared/src/utils/checkTypes.ts:5]: Do we still need this without tscodegen?

**Review by TimoGlastra — APPROVED** [review]
Nice, LGTM. Left some questions

**Comment by blu3beri** [comment]
@TimoGlastra Resolved your feedback.

**Pull request closed without merging**